invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -0,0 +1,244 @@
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Schemaorg based Schema for Invenio RDM Records."""
+from typing import Optional
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from flask_resources.serializers import BaseSerializerSchema
+from pydash import py_
+import pycountry
+from idutils import normalize_orcid, normalize_ror
+from commonmeta import (
+    unwrap,
+    wrap,
+    parse_attributes,
+    dict_to_spdx,
+    doi_as_url,
+)
+from marshmallow import Schema, fields, missing
+from marshmallow_utils.fields import SanitizedUnicode
+
+from ..schemas import CommonFieldsMixin
+from ..utils import get_vocabulary_props
+
+
+class PersonOrOrgSchema(Schema):
+    """Creator/contributor schema for Schemaorg."""
+
+    include = {
+        "@id": fields.Method("get_name_identifier"),
+        "@type": fields.Method("get_name_type"),
+    }
+
+    name = fields.Str(attribute="person_or_org.name")
+    givenName = fields.Str(attribute="person_or_org.given_name")
+    familyName = fields.Str(attribute="person_or_org.family_name")
+    affiliation = fields.Method("get_affiliation")
+
+    def get_name_type(self, obj):
+        """Get name type. Schemaorg expects either @Person or @Organisation.
+        Defaults to @Person.
+        """
+        name_type = py_.get(obj, "person_or_org.type")
+        return "@Organisation" if name_type == "organizational" else "@Person"
+
+    def get_name_identifier(self, obj):
+        """Get name identifier. Schemaorg expects a URL for the identifier, and
+        does not support multiple identifiers. Use the first identifier found.
+        """
+
+        def format_name_identifier(name_identifier) -> Optional[str]:
+            """format_name_identifier"""
+            if name_identifier.get("identifier", None) is None:
+                return None
+            if name_identifier.get("scheme", None) == "orcid":
+                return normalize_orcid(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "ror":
+                return normalize_ror(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "isni":
+                return "http://isni.org/isni/" + name_identifier.get("identifier", None)
+            # TODO: normalize other personororg identifiers, e.g. GND, etc.
+            # Schemaorg expects a URL for the identifier
+            return name_identifier.get("identifier", None)
+
+        return next(
+            (
+                format_name_identifier(i)
+                for i in wrap(obj["person_or_org"].get("identifiers", None))
+            ),
+            None,
+        )
+
+    def get_affiliation(self, obj) -> Optional[list]:
+        """Get affiliation list."""
+
+        def format_affiliation(affiliation) -> Optional[dict]:
+            """format_affiliation"""
+            # TODO: get affiliation id
```

Comment:
Can you check whether this `TODO` can be removed?

Reply:
Just like funding, I left this out in the initial version because of the tight deadline to relaunch Zenodo. I can add at a later time.

Reply:
No worries, I can pick it up and work on it if you'd like. 
Same for funding

Reply:
I am happy to work on what is currently missing after the Zenodo relaunch. This includes a few other things that were not implemented in Zenodo, e.g. geolocation information (supported in DataCite and schema.org).